yt_dlp/extractor/douyutv.py

```diff
@@ -79,28 +95,25 @@ def _real_extract(self, url):
             room_id = self._html_search_regex(
                 r'"room_id\\?"\s*:\s*(\d+),', page, 'room id')
 
-        # Grab metadata from mobile API
+        # Grab metadata from API
+        tt = int(time.time())
+        auth = md5(f'room/{video_id}?aid=wp&client_sys=wp&time={tt}zNzMV1y4EMxOHS6I5WKm'.encode('utf-8')).hexdigest()
         room = self._download_json(
-            'http://m.douyu.com/html5/live?roomId=%s' % room_id, video_id,
-            note='Downloading room info')['data']
+            f'http://www.douyutv.com/api/v1/room/{room_id}', video_id,
+            note='Downloading room info', query={
+                'aid': 'wp',
+                'client_sys': 'wp',
+                'time': tt,
```

Comment:
duplicate code. Why not construct the query object first and use it to generate auth. Like this:
https://github.com/yt-dlp/yt-dlp/blob/074b2fae9076221faaa8697381428131ad968dc9/yt_dlp/extractor/kankanews.py#L29-L38